src/Layouts/Layout.php

```diff
@@ -209,74 +223,107 @@ public function get()
     {
-        $response = [
-            'data' => $this->convertToArray($this->command->execute()->getRecords())
-        ];
-        return $response;
+        $order = $order == 'desc'? FileMaker::SORT_DESCEND : FileMaker::SORT_ASCEND;
+        $this->sortRule = [$key => $order];
+        return $this;
     }
 
     /**
-     * executes the edit command and returns  mapped array of the updated object or throws exception
+     * apply sort rule
+     */
+    public function applySort()
+    {
+        $this->command->addSortRule(key($this->sortRule), 1, current($this->sortRule));
+    }
+    //      END SORTING     //
+
+    //      Filtering     //
+    /**
+     * adds filters to the query
      *
-     * @return array
-     * @throws FieldDoesNotExist
-     * @throws FileMakerException
+     * @param $filters
+     * @param null $value
+     * @return $this
      */
-    public function update()
+    public function filter($filters, $value = null)
     {
-        $response = [
-            'data' => $this->convertToArray($this->command->execute()->getRecords())
-        ];
-        return $response;
+        if (is_string($filters))
+        {
+            $this->filters[]= [$filters => $value];
+        }
+        else{
+            foreach ($filters as $key => $value)
+            {
+                $this->filters[]= [$key => $value];
+            }
+        }
+        return $this;
     }
 
     /**
-     * executes the delete command and returns true or throws exception
+     * applies filters from the filters array to the command
      *
-     * @return array
      */
-    public function delete()
+    protected function applyFilters()
     {
-        $this->command->execute();
-        return [
-            'response' => true
-        ];
+        foreach ($this->filters as $filter)
+        {
+            $this->command->addFindCriterion(key($filter), current($filter));
+        }
     }
+    //      END FILTERING     //
 
+    //      PAGINATION     //
     /**
-     * sets the last record id variable by using pagination with size of 1 to a sorted descending result
+     * adds pagination parameters to the command
      *
-     * @return void
-     * @throws FieldDoesNotExist
-     * @throws FileMakerException
+     * @param $size
+     * @param int $page
+     * @return $this
      */
-    public function setLastRecordId()
+    public function paginate($size, $page = 0)
     {
-        $this->skip_last_record = true;
-        $old_sort_rule = $this->sort_rule;
-        $order = current($this->sort_rule) == FileMaker::SORT_DESCEND? 'asc':'desc';
-        $this->last_record_id =  (int)$this->index()
-            ->sort(key($this->sort_rule), $order)
-            ->paginate(1, 0)
-            ->get()['data'][0][$this->getIdFieldKeyName()];
-        $this->sort_rule = $old_sort_rule;
-        $this->skip_last_record = false;
+        $this->pagination = true;
+        $this->command->setRange($page*$size, $page*$size+$size);
```

Comment:
I don't like depending on default php precedence so use brackets () to make it more readable and may be explain in a simple comment what is going on here for other developers